prototype-spec.md

```diff
@@ -0,0 +1,253 @@
+# Notary v2 Specification
+
+This specification describes the file formats, queries, and processes required for the use of TUF in registries. It requires that metadata for each of the TUF roles: root, snapshot, timestamp, and targets is uploaded to a registry and can be queried by users. Details about these roles can be found in the [design document](https://github.com/notaryproject/nv2/blob/prototype-tuf/tuf-design.md) and the [TUF specification](https://github.com/theupdateframework/specification/blob/master/tuf-spec.md#the-update-framework-specification).
+
+## Root metadata
+
+Root metadata should be signed using an offline process. It can then be uploaded to the registry by running `nv2 upload root`.
+
+## Signing a manifest
+
+To upload a manifest signed with Notary v2, a developer must:
+* Generate and sign targets metadata by running `nv2 sign imagename keyloc --role targetsrolename` where `imagename` is the manifest they are signing, `targetsrolename` is the name of the top-level or delegated targets role the developer is a part of, and `keyloc` is a path to the developer's private key.
```

Comment:
I think we should move away from the idea that private keys are always files sitting on disk. We should generalize this to accommodate hardware keys (like YubiKeys), or HSMs sitting on LAN or even cloud.

Reply:
Agreed. I removed the word 'path', is that better?

Reply:
Yes. We may add concrete examples that uses certain "key server protocols" like `file:/path/to/file` or `kms:key-name` or `gpg:keyid` or `yubikey-piv-slot-9c` and so on, just to convince ourselves that this would work.